### PR TITLE
[draft] lib: example usage and "lib" runmode - v4

### PR DESCRIPTION
**Pull request by jasonish**
For review, and QA purposes and to restart some conversation.

Previous PRs:
- https://github.com/OISF/suricata/pull/11654
- https://github.com/OISF/suricata/pull/10802

Outstanding comments:
- https://github.com/OISF/suricata/pull/10802#discussion_r1560753350
- https://github.com/OISF/suricata/pull/10802#issuecomment-2049317502

Ticket: https://redmine.openinfosecfoundation.org/issues/7240

### CODE REVIEW

File: examples/lib/libcapture/main.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    int datalink = pcap_datalink(fp);
+    int count = 0;
+    struct pcap_pkthdr pkthdr;
+    const u_char *packet;
+    while ((packet = pcap_next(fp, &pkthdr)) != NULL) {
+        if (TmModuleLibHandlePacket(tv, device, packet, datalink, pkthdr.ts, pkthdr.len, 0, 0) !=
+                0) {
+            pthread_exit(NULL);
+        }
```
Comment (jasonish) [inline examples/lib/libcapture/main.c:62]: I'm not thinking that `TmModuleLibHandlePacket` and the call to it should be changed to something like:
- Some helper function to get a packet and perhaps some initial setup. But allow the user to to do some setup as well, like a "release" function.
- User then calls a `TmThreadSlowProcessPkt` or some wrapper around it.
This removes one function from `source-lib.c` making it more functional, as well as allows post-inspection of the packet (which would make sense in workers mode, but not autofp)

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22392

File: examples/lib/libcapture/main.c (hunk 1 above)
Comment (amirabell) [reply examples/lib/libcapture/main.c:62]: Yes not a bad idea.
Will there be an equivalent `TmThreadFastProcessPkt` to `TmThreadSlowProcessPkt`? And what would the difference be?

Comment (jasonish) [reply examples/lib/libcapture/main.c:62]: Oops, typo.. Meant `TmThreadsSlotProcessPkt`.

**Pull request closed without merging**